docs/docs/reference/configuration.md

```diff
@@ -76,6 +76,27 @@ Defines the bootstrap server of Kafka brokers to consume from. Multiple Kafka br
 * Example value: `localhost:9092`
 * Type: `required` only when `KAFKA_CONSUMER` or `KAFKA_SOURCE` is configured in `SOURCE_DETAILS`
 
+##### `SOURCE_KAFKA_CONSUMER_CONFIG_SECURITY_PROTOCOL`
+
+Defines the security protocol used to communicate with ACL enabled kafka. Valid values are: PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL.
+
+* Example value: `SASL_PLAINTEXT`
+* Type: `optional` required only for ACL enabled `KAFKA_CONSUMER` or `KAFKA_SOURCE` 
+
+##### `SOURCE_KAFKA_CONSUMER_CONFIG_SASL_MECHANISM`
+
+Defines the SASL mechanism used for client connections with ACL enabled kafka. This may be any mechanism for which a security provider is available.
+
+* Example value: `SCRAM-SHA-512`
+* Type: Type: `optional` required only for ACL enabled `KAFKA_CONSUMER` or `KAFKA_SOURCE`
+
+##### `SOURCE_KAFKA_CONSUMER_CONFIG_SASL_JAAS_CONFIG`
+
+Defines the SASL JAAS Config used for JAAS login context parameters for SASL connections in the format used by JAAS configuration files. JAAS configuration file format is described [here](http://docs.oracle.com/javase/8/docs/technotes/guides/security/jgss/tutorials/LoginConfigFile.html).
+
+* Example value: `loginModuleClass controlFlag (optionName=optionValue)*`
+* Type: `optional`
```

Comment:
Can we mention when does the user need to provide this ?
For what source ? Also, are there any dependent configs for this ?